packages/live-share-inking/src/core/InkingManager.ts

```diff
@@ -0,0 +1,1293 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the Microsoft Live Share SDK License.
+ */
+
+import { EventEmitter } from "events";
+import { CanvasReferencePoint, InkingCanvas } from "../canvas/InkingCanvas";
+import { DryCanvas, WetCanvas } from "../canvas/DryWetCanvas";
+import { LaserPointerCanvas } from "../canvas/LaserPointerCanvas";
+import { IPoint, IPointerPoint, IRect, makeRectangle, screenToViewport, viewportToScreen } from "./Geometry";
+import { Stroke, IStroke, IStrokeCreationOptions, StrokeType } from "./Stroke";
+import { InputFilter, InputFilterCollection } from "../input/InputFilter";
+import { JitterFilter } from "../input/JitterFilter";
+import { generateUniqueId, pointerEventToPoint } from "./Utils";
+import { InputProvider } from "../input/InputProvider";
+import { PointerInputProvider } from "../input/PointerInputProvider";
+import { DefaultHighlighterBrush, DefaultLaserPointerBrush, DefaultPenBrush, IBrush } from "./Brush";
+
+interface CoalescedPointerEvent extends PointerEvent {
+    getCoalescedEvents(): PointerEvent[];
+}
+
+/**
+ * @hidden
+ * Retrieves a list of coalesced PointerEvent objects from a single PointerEvent.
+ * PointerEvent.getCoalescedEvents() is an experimental feature and TypeScript doesn't
+ * yet declare it.
+ * For more info: https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/getCoalescedEvents
+ * @param event The PointerEvent to get coalesced events from.
+ * @returns An array of coalesced PointerEvent objects.
+ */
+function getCoalescedEvents(event: PointerEvent): PointerEvent[] {
+    if ('getCoalescedEvents' in event) {
+        const events: PointerEvent[] = (event as CoalescedPointerEvent).getCoalescedEvents();
+
+        // Firefox can return an empty list. See https://bugzilla.mozilla.org/show_bug.cgi?id=1511231.
+        if (events.length >= 1) {
+            return events;
+        }
+    }
+
+    return [event];
+}
+
+/**
+ * Defines available inking tools.
+ */
+export enum InkingTool {
```

Comment:
>InkingTool [](http://example.com/codeflow?start=12&length=10)

Just want to make sure we're ok with all inking tools having to be defined by us since this is a closed list.

Reply:
I have a plan for an extensibility model, but I think it goes beyond the scope for now. When we're ready for it, I can easily maintain backward compat with this enum.